Fix:

TypeError: cannot use a string pattern on a bytes-like object:
https://github.com/apache/serf/commit/9d30108b630b77f732ef94d1642b159066ffd890

SyntaxError: Missing parentheses in call to 'print'. Did you mean print(...)?
https://github.com/apache/serf/commit/1c0241847e845a8da265aa89f297d3db6ae483b1
--- SConstruct.orig	2015-09-17 07:46:24.000000000 -0500
+++ SConstruct	2022-08-09 18:04:36.000000000 -0500
@@ -167,7 +167,7 @@
 match = re.search('SERF_MAJOR_VERSION ([0-9]+).*'
                   'SERF_MINOR_VERSION ([0-9]+).*'
                   'SERF_PATCH_VERSION ([0-9]+)',
-                  env.File('serf.h').get_contents(),
+                  env.File('serf.h').get_contents().decode('utf-8'),
                   re.DOTALL)
 MAJOR, MINOR, PATCH = [int(x) for x in match.groups()]
 env.Append(MAJOR=str(MAJOR))
@@ -184,7 +184,7 @@
 
 unknown = opts.UnknownVariables()
 if unknown:
-  print 'Warning: Used unknown variables:', ', '.join(unknown.keys())
+  print('Warning: Used unknown variables:', ', '.join(unknown.keys()))
 
 apr = str(env['APR'])
 apu = str(env['APU'])
